app/src/main/java/com/concordium/wallet/ui/cis2/lookfornew/SelectTokensFragment.kt

```diff
@@ -145,13 +149,15 @@ class SelectTokensFragment : TokensBaseFragment() {
                 binding.updateWithTokens.text = getString(R.string.cis_update_tokens)
             else
                 binding.updateWithTokens.text = getString(R.string.cis_add_tokens)
-            binding.updateWithTokens.isEnabled = true
         }
         _viewModel.nonSelected.observe(viewLifecycleOwner) { nonSelected ->
             if (nonSelected)
                 binding.nonSelected.visibility = View.VISIBLE
             else
                 binding.nonSelected.visibility = View.INVISIBLE
         }
+        _viewModel.selectedTokens.observe(viewLifecycleOwner) {
+            binding.updateWithTokens.isEnabled = it.isNotEmpty()
+        }
```

Comment:
I'm not sure this is the correct fix (I'm not sure I agree with the issue itself)
It should be possible to use this page to remove all tokens in the contract, I believe :thinking:

Reply:
@orhoj

Reply:
Yes, we can't do it like that as it is also valid to de-select all tokens and then do the update. In the browser wallet the button is always just active and will tell you that no updates were made if you update with no changes.

Alternatively it should track if any changes have been made, and then only activate the button at that point. But then the applications would be doing it differently which is bad.

So I guess I disagree with the ticket in this being a problem.